autoscheduler/frontend/src/components/ScheduleCard/ScheduleCard.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import * as styles from './ScheduleCard.css';
+import DragHandle from './DragHandle';
+import { RootState } from '../../redux/reducers';
+import { AvailabilityArgs } from '../../types/Availability';
+
+let contentHeight: number = null;
+
+interface BasicProps {
+  startTimeHours: number;
+  startTimeMinutes: number;
+  endTimeHours: number;
+  endTimeMinutes: number;
+  borderColor?: string;
+  backgroundColor?: string;
+  backgroundStripes?: boolean;
+  firstHour: number;
+  lastHour: number;
+  onResizeWindow?: (isBig: boolean) => void;
+  onDragHandleDown?: (evt: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    endSelected: boolean) => void;
+}
+
+type ScheduleCardProps = React.PropsWithChildren<BasicProps>;
+
+const ScheduleCard: React.FC<ScheduleCardProps> = ({
+  startTimeHours, startTimeMinutes, endTimeHours, endTimeMinutes,
+  borderColor, backgroundColor, backgroundStripes, firstHour, lastHour, children,
+  onResizeWindow, onDragHandleDown,
+}) => {
+  const selectedAvailability = useSelector<RootState, AvailabilityArgs>(
+    (state) => state.selectedAvailability,
+  );
+  // tracks height of card and content, hiding meeting type if necessary
+  const [isHovered, setHovered] = React.useState(false);
+  const [isMouseDown, setMouseDown] = React.useState(!!selectedAvailability);
+  const cardRoot = React.useRef<HTMLDivElement>(null);
+  const cardContent = React.useRef<HTMLDivElement>(null);
+  React.useEffect(() => {
+    const handleResize = (): void => {
+      // set initial height for future use
+      contentHeight = contentHeight || cardContent.current.clientHeight;
+
+      if (contentHeight >= cardRoot.current.clientHeight) {
+        if (onResizeWindow) onResizeWindow(false);
+      } else if (onResizeWindow) onResizeWindow(true);
```

Comment:
```suggestion
      if (onResizeWindow) {
        onResizeWindow(contentHeight < cardRoot.current.clientHeight);
      }
```
I think this is functionally identical, the way it's written right now (checking if `onResizeWindow` in a different place depending on the branch) is a little confusing. Also could use a comment explaining what the bool value represents (it's hard to tell without looking at `MeetingCard` since that's the only component that currently uses this value)

Reply:
Naw you right